src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
+            } else {
+                func_8002EBCC(&this->actor, globalCtx, 0);
+                func_80093D18(globalCtx->state.gfxCtx);
+                gDPSetRenderMode(POLY_OPA_DISP++, G_RM_PASS, G_RM_AA_ZB_OPA_SURF2);
+                Matrix_RotateY(this->triforceSpot.rotation * (M_PI / 32768.0f), MTXMODE_APPLY);
+                gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_demo_effect.c", 3085),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPSegment(POLY_OPA_DISP++, 8,
+                           Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 32, 16, 1, 0, 0, 16, 8));
+                gDPSetPrimColor(POLY_OPA_DISP++, 0x80, 0x80, 0xFF, 0xFF, 0xA0, 0xFF);
+                gDPSetEnvColor(POLY_OPA_DISP++, 0xAA, 0x8C, 0x00, 0xFF);
+                gSPDisplayList(POLY_OPA_DISP++, triforceSpot);
+            }
+        }
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_demo_effect.c", 3112);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_809746B4.s")
+/**
+ * Draw function for the Get Item Actors.
+ * This is either Medals or Light Arrows based on the giIndex.
+ */
+void DemoEffect_DrawGetItem(DemoEffect* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    if (!DemoEffect_CheckCsAction(this, globalCtx, 1) && !DemoEffect_CheckCsAction(this, globalCtx, 4)) {
+        if (!this->getItem.isLoaded) {
+            this->getItem.isLoaded = 1;
+            return;
+        }
+        func_8002EBCC(thisx, globalCtx, 0);
+        func_8002ED80(thisx, globalCtx, 0);
+        func_800694A0(globalCtx, this->getItem.giIndex);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_809746F8.s")
+/**
+ * Callback for the SkelCurve system to draw the animated limbs.
+ */
+s32 DemoEffect_DrawTimewarpLimbs(GlobalContext* globalCtx, SkelAnimeCurve* skelCuve, s32 limbIndex, Actor* thisx) {
+    s32 pad;
+    DemoEffect* this = THIS;
+    u32 scroll = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_80974B80.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_demo_effect.c", 3154);
+    func_80093D84(globalCtx->state.gfxCtx);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x00, 0x80, 0xAA, 0xFF, 0xFF, 0xFF);
+    gDPSetEnvColor(POLY_XLU_DISP++, this->envXluColor[0], this->envXluColor[1], this->envXluColor[2], 0xFF);
+    gSPSegment(POLY_XLU_DISP++, 8,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (scroll * 6) % 1024, 256 - ((scroll * 16) % 256) - 1, 256,
+                                64, 1, (scroll * 4) % 512, 128 - ((scroll * 12) % 128) - 1, 128, 32));
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_demo_effect.c", 3172);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_80974EB4.s")
+    if (limbIndex == 0) {
+        LimbTransform* transforms = skelCuve->transforms;
+        transforms->scale.x = 1024;
```

Comment:
Chained assignment?

Reply:
NOT OK :D

Reply:
Even if you play around with the order? That's a bit surprising

Reply:
y = 1024
z = x = 1024 


was the only way i got this to match lol. its better than setting it to the var tho imo